agent/session.go

```diff
@@ -31,26 +31,26 @@ type session struct {
 	conn *grpc.ClientConn
 	addr string
 
-	agent     *Agent
-	sessionID string
-	session   api.Dispatcher_SessionClient
-	errs      chan error
-	messages  chan *api.SessionMessage
-	tasks     chan *api.TasksMessage
+	agent       *Agent
+	sessionID   string
+	session     api.Dispatcher_SessionClient
+	errs        chan error
+	messages    chan *api.SessionMessage
+	assignments chan *api.AssignmentsMessage
```

Comment:
We could remove Assign, but we would have to pass in a parameter to Update that informs Update if this is a COMPLETE or not, to make sure to remove dangling tasks on a re-sync.